src/System.IO.Pipelines/System/Buffers/MemoryPool.cs

```diff
@@ -12,39 +12,34 @@ namespace System.Buffers
     /// </summary>
     public class MemoryPool : MemoryPool<byte>
     {
+#if NETSTANDARD1_1
         /// <summary>
         /// The gap between blocks' starting address. 4096 is chosen because most operating systems are 4k pages in size and alignment.
         /// </summary>
-        private const int _blockStride = 4096;
-
+        private const int _pageSize = 4096;
+#else
         /// <summary>
-        /// The last 64 bytes of a block are unused to prevent CPU from pre-fetching the next 64 byte into it's memory cache.
-        /// See https://github.com/aspnet/KestrelHttpServer/issues/117 and https://www.youtube.com/watch?v=L7zSU9HI-6I
+        /// The gap between blocks' starting address.
         /// </summary>
-        private const int _blockUnused = 64;
-
+        private static readonly int _pageSize = Environment.SystemPageSize;
+#endif
         /// <summary>
         /// Allocating 32 contiguous blocks per slab makes the slab size 128k. This is larger than the 85k size which will place the memory
         /// in the large object heap. This means the GC will not try to relocate this array, so the fact it remains pinned does not negatively
         /// affect memory management's compactification.
         /// </summary>
         private const int _blockCount = 32;
 
-        /// <summary>
-        /// 4096 - 64 gives you a blockLength of 4032 usable bytes per block.
-        /// </summary>
-        private const int _blockLength = _blockStride - _blockUnused;
-
         /// <summary>
         /// Max allocation block size for pooled blocks,
         /// larger values can be leased but they will be disposed after use rather than returned to the pool.
         /// </summary>
-        public override int MaxBufferSize { get; } = _blockLength;
+        public override int MaxBufferSize { get; } = _pageSize;
 
         /// <summary>
         /// 4096 * 32 gives you a slabLength of 128k contiguous bytes allocated per slab
         /// </summary>
-        private const int _slabLength = _blockStride * _blockCount;
+        private static readonly int _slabLength = _pageSize * _blockCount;
```

Comment:
What are the possible values for _pageSize? I assume it will never be less than 4096, but I want to be sure of this so we don't end up with a _slabLength so short that it doesn't end up in the LOH. We should probably update some comments that assume a 4096 block stride.

Reply:
Page size is always a power of 2 as its integral for determining what page an address is in (OS/CPU)

Don't know if could be less than 4096; perhaps
```csharp
private static readonly int _pageSize = Math.Max(4096, Environment.SystemPageSize);
``` 
Would still be valid for pages sizes less than 4096. However, might want to also adjust number of blocks; incase someone has set page size to 2MB

Reply:
Huge pages may become more of a thing on linux due to Meltdown; so should cap the multiplier so it doesn't allocate a vast amount per slab